knowledge-base/how-to-fix-axe-errors-for-html5-viewer.md

```diff
@@ -0,0 +1,110 @@
+---
+title: How to fix Axe accessibility violations of HTML5 report viewer
+description: How to fix errors reported by Axe on the page that contains HTML5 report viewer
+type: how-to
+page_title: Fix Axe Accessibility Violations of HTML5 Report Viewer
+slug: how-to-fix-axe-errors-for-html5-viewer
+position: 
+tags: accessibility,html5reportviewer
+ticketid: 1451834
+res_type: kb
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product Version</td>
+	    	<td>11.2.17.913 or higher</td>
+	    </tr>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Progress® Telerik® Reporting</td>
+	    </tr>
+    </tbody>
+</table>
+
+
+## Description
+[Axe web accessibility testing tool](https://www.deque.com/axe/) might report violations and warnings releated to HTML5 Report Viewer and the content of the report.
+
+When providing accessibility for the report viewer we were guided by [Section 508 of the Rehabilitation Act](http://www.section508.gov/), 
+[Web Content Accessibility Guidelines (WCAG) 2.0](https://www.w3.org/TR/WCAG20/), and the standards set by [WAI-ARIA](https://www.w3.org/TR/wai-aria-practices/). 
+The implemented approach was tested using [Microsoft Narrator](https://en.wikipedia.org/wiki/Microsoft_Narrator) and [JAWS](https://www.freedomscientific.com/products/software/jaws/)
+screen readers.
+
+As there is no common stardard for implementing the accessibility for web applications some of the accessibility testing tools might 
+have additional guidelinies for implementing accessibility which are not respected by HTML5 Report Viewer.
+
+## Error Messages
+
+Elements should not have tabindex greater than zero
+
+Certain ARIA roles must contain particular children
+
+Elements must have sufficient color contrast
+
+
+## Solution
+
+In Telerik Reporting R3 2017 (version 11.2.17.913) we introduced HTML5 Report Viewer accessibility affecting both the viewer controls 
+and the report content. The accessibility features are enabled or disabled using *enableAccessibility* option of the viewer
+(its default value is false).
+
+As a first step, make sure that accessibility is enabled for the viewer as this step will help to eliminate most of the reported
+warnings and violations:
+
+```JavaScript
+$("#reportViewer1")
+                .telerik_ReportViewer({
+                        enableAccessibility: true,
+```
+
+Below are the solutions to eliminate the remaining violations reported by Axe.
+
+*Elements should not have tabindex greater than zero*
+
+The viewer can comply with this requirement by having all tabindex attributes set to 0. Changing the value of all tabindex attributes 
```

Comment:
I agree with you that it's a good idea to explain why we enforce this tabindex. 
I checked its values in the reportviewer template and the order is this: Menu, DocumentMap, Parameters Area, Search dialog, Send Email dialog. Of course, the order is respected only for the visible targets.
Additionally in the report content we enforce the tabindex of the report items, so each textbox, picture, shape, table, graph, map, etc. will have the same order as designed, i.e. the natural order of items is preserved.